### PR TITLE
nrf_security: User supplied PRNG

### PR DESCRIPTION
**Pull request by vili-nordic**
Add option to allow application to provide random data.

ref: NCSDK-21958

### CODE REVIEW

File: subsys/nrf_security/Kconfig.psa [hunk 1]
```diff
@@ -88,6 +88,11 @@ config PSA_ITS_ENCRYPTED
 	help
 	  Enables authenticated encryption for PSA Internal Trusted Storage files
 
+config PSA_CRYPTO_DRIVER_ALG_PRNG_TEST
+	bool
+	prompt "Application provided PRNG"
```
Comment (vili-nordic) [inline subsys/nrf_security/Kconfig.psa:93]: ticket says promptless, but I can't set the option in prj.conf then.

Unsure how this was intended

**Review by mswarowsky — APPROVED** [review]

**Comment by vili-nordic** [comment]
Another question is what we will do with TF-M. Users/testers can't really supply their function to be linked in there?

File: subsys/nrf_security/Kconfig.psa [hunk 2]
```diff
@@ -88,6 +88,10 @@ config PSA_ITS_ENCRYPTED
 	help
 	  Enables authenticated encryption for PSA Internal Trusted Storage files
 
+config PSA_CRYPTO_DRIVER_ALG_PRNG_TEST
+	bool
+	help
+		Internal option used for testing.
```
Comment (Vge0rge) [inline subsys/nrf_security/Kconfig.psa:94]: Nit: You can expand just a bit here if you want.

**Review by Vge0rge — APPROVED** [review]

File: subsys/nrf_security/Kconfig.psa (hunk 2 above)
Comment (joerchan) [reply subsys/nrf_security/Kconfig.psa:94]: More nit: Indent: 1 tab + 2 spaces

**Comment by SebastianBoe** [comment]
"Another question is what we will do with TF-M. Users/testers can't really supply their function to be linked in there?"

TF-M users can supply application-specific CMake variables.
One option would be to supply a static inline header implementation as is done for user-specific mbedtls configs.

So from driver wrapper do:

#ifdef TFM_TEST_PRNG_HEADER
#include TFM_TEST_PRNG_HEADER
#endif

And from TF-M do

if(TFM_TEST_PRNG_HEADER)
add_definitions(TFM_TEST_PRNG_HEADER=${TFM_TEST_PRNG_HEADER})
endif()

Or something like this.

**Review by SebastianBoe — APPROVED** [review]

**Pull request merged**